pallets/phala-world/src/nft_sale.rs

```diff
@@ -0,0 +1,1722 @@
+		/// reduce the time left to incubation if the origin_of_shell is in the top 10 of
+		/// origin_of_shells fed that era.
+		///
+		/// Parameters:
+		/// - origin: The origin of the extrinsic feeding the origin_of_shell
+		/// - collection_id: The collection id of the Origin of Shell RMRK NFT
+		/// - nft_id: The NFT id of the Origin of Shell RMRK NFT
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn feed_origin_of_shell(
+			origin: OriginFor<T>,
+			collection_id: CollectionId,
+			nft_id: NftId,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+			// TODO: Move to incubation.rs
+
+			Ok(())
+		}
+
+		/// Hatch the origin_of_shell that is currently being hatched. This will trigger the end of
+		/// the incubation process and the origin_of_shell will be burned. After burning, the user
+		/// will receive the awakened Shell RMRK NFT
+		///
+		/// Parameters:
+		/// - origin: The origin of the extrinsic incubation the origin_of_shell
+		/// - collection_id: The collection id of the Origin of Shell RMRK NFT
+		/// - nft_id: The NFT id of the Origin of Shell RMRK NFT
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn hatch_origin_of_shell(
+			origin: OriginFor<T>,
+			collection_id: CollectionId,
+			nft_id: NftId,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+			// TODO: Move to incubation.rs
+
+			Ok(())
+		}
+
+		/// This is an admin function to update origin_of_shells incubation times based on being in
+		/// the top 10 of fed origin_of_shells within that era
+		///
+		/// Parameters:
+		/// - origin: The origin of the extrinsic updating the origin_of_shells incubation times
+		/// - collection_id: The collection id of the Origin of Shell RMRK NFT
+		/// - nft_id: The NFT id of the Origin of Shell RMRK NFT
+		/// - reduced_time: The amount of time the origin_of_shell will be reduced by
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn update_incubation_time(
+			origin: OriginFor<T>,
+			collection_id: CollectionId,
+			nft_id: NftId,
+			reduced_time: u64,
+		) -> DispatchResult {
+			// Ensure OverlordOrigin makes call
+			let sender = ensure_signed(origin)?;
+			Self::ensure_overlord(sender)?;
+			// TODO: Move to incubation.rs
+
+			Ok(())
+		}
+
+		/// Privileged function set the Overlord Admin account of Phala World
+		///
+		/// Parameters:
+		/// - origin: Expected to be called by `OverlordOrigin`
+		/// - new_overlord: T::AccountId
+		#[pallet::weight(0)]
+		pub fn set_overlord(
+			origin: OriginFor<T>,
+			new_overlord: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			// This is a public call, so we ensure that the origin is some signed account.
+			T::OverlordOrigin::ensure_origin(origin)?;
+			let old_overlord = <Overlord<T>>::get();
+
+			Overlord::<T>::put(&new_overlord);
+			Self::deposit_event(Event::OverlordChanged { old_overlord });
```

Comment:
Better to also include the new overlord

Reply:
I was thinking we didn't want the Account ID of the new overlord emitted as an Event to keep the public key from being exposed. I can include the new overlord if that's best.